snap/info_package_yaml.go

```diff
@@ -0,0 +1,38 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package snap
+
+import (
+	"fmt"
+	"gopkg.in/yaml.v2"
+)
+
+// NewFromPackageYaml creates a new info based on the given packageYaml
+func NewFromPackageYaml(yamlData []byte) (*Info, error) {
```

Comment:
This should probably be `snap.InfoFromPackageYaml` now that it sits under the `snap` package (it's not a "new snap").